collector/collector.go

```diff
@@ -0,0 +1,37 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package collector handles the command-line, configuration, and runs the
+// OpenTelemetry Collector.
+package collector // import "go.opentelemetry.io/collector/collector"
```

Comment:
Any thoughts on an alternative name? The repetition `collector/collector` needs improving.

Reply:
`otelcol`?

Reply:
`otelcol` is ok, no names that come to mind are any better.

Reply:
Should the type be `otelcol.Collector` or `otelcol.OtelCol` or `otelcol.Otelcol`.

Otherwise, if we stick with `otelcol.Collector` the `otelcol.New` I believe (for consistency with other places) should be `otelcol.NewCollector`

Reply:
@codeboten ping on this :)

Reply:
`otelcol.Collector` and `otelcol.NewCollector` make sense to me 👍🏻

Reply:
Done.